packages/block-library/src/playlist/save.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	RichText,
+	useBlockProps,
+	useInnerBlocksProps,
+	__experimentalGetElementClassName,
+} from '@wordpress/block-editor';
+
+export default function saveWithInnerBlocks( { attributes } ) {
+	const {
+		caption,
+		showNumbers,
+		showTracklist,
+		showArtists,
+		tagName: TagName = showNumbers ? 'ol' : 'ul',
+	} = attributes;
+
+	const blockProps = useBlockProps.save();
+	const innerBlocksProps = useInnerBlocksProps.save( blockProps );
+	return (
+		<figure { ...innerBlocksProps }>
+			<TagName
+				className={
+					'wp-block-playlist__tracklist' +
+					( ! showTracklist
+						? ' wp-block-playlist__tracklist-is-hidden'
+						: '' ) +
+					( ! showArtists
+						? ' wp-block-playlist__tracklist-artist-is-hidden'
+						: '' )
+				}
```

Comment:
Let's avoid concatenating class names as strings, as this can introduce unintended spaces.
```suggestion
				className={ clsx( 'wp-block-playlist__tracklist', {
					'wp-block-playlist__tracklist-is-hidden': ! showTracklist,
					'wp-block-playlist__tracklist-artist-is-hidden':
						! showArtists,
				} ) }
```